### PR TITLE
net: dhcpv4: add support to set dhcpv4 option 60

### PR DESCRIPTION
**Pull request by maass-hamburg**
Adds the funktion to set the DHCPv4 option 60
(Vendor class identifier) via Kconfig.

### CODE REVIEW

File: subsys/net/lib/dhcpv4/Kconfig [hunk 1]
```diff
@@ -44,6 +44,22 @@ config NET_DHCPV4_ACCEPT_UNICAST
 	  If set, the network stack will accept unicast DHCPv4 responses from
```
Comment (jukkar) [inline subsys/net/lib/dhcpv4/Kconfig:44]: Could you modify the commit subject, the option 60 does not tell anything and no need to mention dhcpv4 twice.
Perhaps something like `net: dhcpv4: Add vendor class identifier option`

File: subsys/net/lib/dhcpv4/Kconfig [hunk 2]
```diff
@@ -44,6 +44,22 @@ config NET_DHCPV4_ACCEPT_UNICAST
...
 	  servers, before the assigned address is configured on the interface.
 
+config NET_DHCPV4_VENDOR_CLASS_IDENTIFIER
+	bool "Send vendor class identifier in DHCPv4 request"
+	depends on NET_DHCPV4
+	default n
```
Comment (jukkar) [inline subsys/net/lib/dhcpv4/Kconfig:50]: Option are off by default so this line can be omitted.

File: subsys/net/lib/dhcpv4/Kconfig [hunk 3]
```diff
@@ -44,6 +44,22 @@ config NET_DHCPV4_ACCEPT_UNICAST
...
+	help
+		If set, the network stack will include the specified string in the
+		DHCPv4 vendor class identifier option in the DHCPv4 request.
```
Comment (jukkar) [inline subsys/net/lib/dhcpv4/Kconfig:53]: Please indent the same way as other help options, there should be two space indentation for the help string.

File: subsys/net/lib/dhcpv4/dhcpv4.c [hunk 4]
```diff
@@ -164,6 +173,10 @@ static struct net_pkt *dhcpv4_create_message(struct net_if *iface, uint8_t type,
 	const char *hostname = net_hostname_get();
 	const size_t hostname_size = strlen(hostname);
 #endif
+#if defined(CONFIG_NET_DHCPV4_VENDOR_CLASS_IDENTIFIER)
+	const char *vendor_class_id = CONFIG_NET_DHCPV4_VENDOR_CLASS_IDENTIFIER_STRING;
+	const size_t vendor_class_id_size = strlen(vendor_class_id);
```
Comment (jukkar) [inline subsys/net/lib/dhcpv4/dhcpv4.c:178]: We could avoid runtime call to `strlen` if done like this:
```
	const char vendor_class_id[] = CONFIG_NET_DHCPV4_VENDOR_CLASS_IDENTIFIER_STRING;
	const size_t vendor_class_id_size = sizeof(vendor_class_id) - 1;
```

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**